nibabel/tests/test_proxy_api.py

```diff
@@ -262,7 +266,7 @@ def sio_func():
                 dtype=dtype,
                 dtype_out=dtype_out,
                 arr=arr.copy(),
-                arr_out=arr * slope + inter,
+                arr_out=arr.astype(dtype_out) * slope + inter,
```

Comment:
If `arr` is float32, `arr * slope + inter` is float32. This hasn't mattered with the simple factors we've been using, but the new parameters produce different rounding errors in float32 (basic numpy coercion) and float64 (nibabel array scaling).

This change ensures that we produce our expected array in the wider dtype.